### PR TITLE
doc: split preprocessing into ref and explanation

### PR DESCRIPTION
**Pull request by emillon**
The existing section from "concepts" mixes reference info and an explanation about how preprocessing works. These concerns are split into two pages.

### CODE REVIEW

File: doc/reference/preprocessing-spec.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+     (with-stdout-to %{target}
+      (chdir %{workspace_root} <action>))))
+
+The equivalent of a ``-pp <command>`` option passed to the OCaml compiler is
+``(system "<command> %{input-file}")``.
+
+Using PPX Rewriters
+-------------------
+
+If ``(pps <ppx-rewriters-and-flags>)`` is used, the corresponding rewriters are set up using the "fast pipeline" (using a separate preprocessing step).
```
Comment (Alizter) [inline doc/reference/preprocessing-spec.rst:52]: Could you wrap these lines?

**Review by christinerose — APPROVED** [review]

File: doc/reference/preprocessing-spec.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+The distinction between these pipelines is explained in
+:doc:`../explanation/preprocessing`.
+
+Arguments to PPX Rewriters
+--------------------------
+
+In ``(pps <ppx-rewriters-and-flags>)`` and ``(staged_pps
+<ppx-rewriters-and-flags>)``, ``<ppx-rewriters-and-flags>`` is a sequence where
+each element is either a command line flag if starting with a ``-`` or the name
```
Comment (Alizter) [inline doc/reference/preprocessing-spec.rst:65]:
```suggestion
each element is either a command line flag if it starts with a ``-`` or the name
```

File: doc/reference/preprocessing-spec.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+--------------------------------------
+
+By default, a preprocessing specification applies to all modules in the
+library/set of executables. It's possible to select the preprocessing on a
+module-by-module basis by using the ``(per-module ...)`` syntax. For instance::
+
+  (preprocess
+   (per_module
+    (((action (run ./pp.sh X=1 %{input-file})) foo bar))
+    (((action (run ./pp.sh X=2 %{input-file})) baz))))
```
Comment (Alizter) [inline doc/reference/preprocessing-spec.rst:105]: It seems like there are too many brackets here. Is this really the case?

File: doc/explanation/preprocessing.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
+How Preprocessing Works
+=======================
+
+Preprocessing consists in transforming source code before it is compiled. The
```
[the remark this answers is not in the document]
Comment (emillon) [reply doc/explanation/preprocessing.rst:4]: as I understand it, "consists in" = "is the process of", "consists of" = "is composed of"

[the remark this answers is not in the document]
Comment (Alizter) [reply doc/explanation/preprocessing.rst:4]: OK but it sounds strange to my (British) English ear.

**Review by Alizter — APPROVED** [review]

**Pull request merged**